packages/Popover/src/Popover.js

```diff
@@ -244,7 +249,8 @@ class Popover extends React.Component {
   }, throttleDelay);
 
   handleKeyUp = event => {
-    if (event.key === "Escape") {
+    if (event.key === "Escape" && isDescendantOfActiveElement()) {
```

Comment:
Seeing an inconsistency between this and the overlay component above in checking `isOpen`. Kinda makes sense to check for it, but also likely impossible to get here if it's not open. Up to you whether to change it or not